src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -89,6 +89,22 @@ public static int Main(string[] args)
                 Environment.Version,
                 RuntimeFramework.CurrentFramework.DisplayName);
 
+            // Restore the COMPLUS_Version env variable if it's been overridden by TestAgency::LaunchAgentProcess
+            try
+            {
+              string cpvOriginal = Environment.GetEnvironmentVariable("TestAgency_COMPLUS_Version_Original");
+              if(!string.IsNullOrEmpty(cpvOriginal))
+              {
+                log.Debug("Agent process has the COMPLUS_Version environment variable value \"{0}\" overridden with \"{1}\", restoring the original value.", cpvOriginal, Environment.GetEnvironmentVariable("COMPLUS_Version"));
```

Comment:
I'm not sure about the ways an agent might be run.
For instance, on mono this won't be set, as it's gotten different ways for enforcing the CLR version (or I've heard Mono has dropped anything but the recent one).